### DISCUSSION TITLE
Album sort splits per year

### DISCUSSION THREAD

**Discussion by GK-774** [jcorporation/myMPD#1203] [Q&A] [answered]
Version 13.0.6 (current stable)

If an album has songs tagged with different years (but album title, album artist and/or artist identical) it is shown as different albums (per year). This is wrong, an album with one album title and one artist is one album. How to fix that?

**Comment by jcorporation** [comment]
This is the new standard, but it can be configured. It is documented here: https://jcorporation.github.io/myMPD/references/tags. Simply change the Date tag in this file and restart myMPD.

**Reply by GK-774** [reply]
Well, "new standard", which kind of standard is this? It´s not unusual for an album to contain tracks that are not recorded in the same year. That is no standard, it´s just plain wrong. Anyway, glad to hear it can be configured.
So if I set that to "AlbumArtist" will it fall back then to "Artist" if the "AlbumArtist" tag is not present? (That is the behaviour I want to achieve, "AlbumArtist" for compilations, "Artist" for usual albums with one artist, band or orchestra).

**Reply by jcorporation** [reply, marked as answer]
> Well, "new standard", which kind of standard is this?

This means only the default setting of myMPD. As I know MALP uses the same default to group albums.

The fallback behavior from AlbumArtist to Artist is not changed. Simply replace "Date" with "Unknown" in the described file.

**Reply by GK-774** [reply]
Just out of couriosity, why set it to "Unknown", setting "AlbumArtist" would be wrong?
Edit: Yes it would be wrong. Results in "Album not found" errors.

**Reply by GK-774** [reply]
BTW, setting album_mode to "simple" is supposed to work? For me it just results in mympd hanging.

**Reply by jcorporation** [reply]
> Just out of couriosity, why set it to "Unknown", setting "AlbumArtist" would be wrong?

Album + (Album)Artist is always used. The setting configures an additional grouping tag.

> BTW, setting album_mode to "simple" is supposed to work? For me it just results in mympd hanging.

Yes, simple album_mode should work.

**Reply by GK-774** [reply]
Well, it does not. After setting simple album_mode no albums at all are shown, did a "Clear app cache and reload" which results in mympd hanging (can´t be stopped via systemctl and stopped logging, had to do a kill -9). Removed album_mode and everything is back to normal.
[mympd_log.zip](https://github.com/jcorporation/myMPD/files/13919411/mympd_log.zip)

**Reply by jcorporation** [reply]
I can not reproduce it. Can you provide myMPD debug and MPD verbose logs?

**Comment by GK-774** [comment]
Log files as requested.
[mpd_log_verbose.zip](https://github.com/jcorporation/myMPD/files/13921023/mpd_log_verbose.zip)
[mympd_log_debug.zip](https://github.com/jcorporation/myMPD/files/13921024/mympd_log_debug.zip)

At the end killed it because systemctl could not stop it:
[root@mpd8 ~]# ps -fade | grep mympd
mympd     173615       1 94 16:37 ?        00:02:33 /usr/bin/mympd
root      173616  173404  0 16:37 pts/0    00:00:00 journalctl -f -u mympd
root      173687  173645  0 16:39 pts/2    00:00:00 systemctl stop mympd.service
root      173736  173693  0 16:40 pts/3    00:00:00 grep --color=auto mympd
[root@mpd8 ~]# kill -9 173615

**Reply by jcorporation** [reply]
The MYMPD_API_DATABASE_ALBUM_LIST API call does not return and that's why myMPD must be killed. Is the cpu usage of the mympd process very high after you go to the album view?

Edit:
I think I found the error. The sort tag of the MYMPD_API_DATABASE_ALBUM_LIST api call is set to Last-Modified. This field is not populated in the simple album mode. This causes the album sort loop to run for each album long and longer.

**Comment by GK-774** [comment]
Thanks for the information. Yes, "Last-Modified" is what I use and as I understand now, this would not work with simple album_mode (which I tried just out of couriousity) because it would not provide the required information for this kind of sorting. Thanks for the fix in devel.
